src/aqua/installation-spell/src/aqua/cli.aqua

```diff
@@ -13,52 +13,66 @@ func upload_deploy(config: LocalAppDeployConfig) -> DeployedAppWorkers:
     deployed <- deploy(def)
     <- deployed
 
-data LogDirect:
+data LogsDirectResult:
+    success: bool
+    error: ?string
     spell_id: string
     worker_id: string
     host_id: string
     worker_name: string
     logs: []Log
 
-func get_logs(app_workers: DeployedAppWorkers) -> *LogDirect:
-    logs: *LogDirect
+func get_logs(app_workers: DeployedAppWorkers) -> []LogsDirectResult:
+    logs: *LogsDirectResult
     for w <- app_workers.workers:
         for i <- w.installation_spells:
             on i.worker_id via i.host_id:
                 Spell i.spell_id
                 get_res <- Spell.get_logs()
-                logs <<- LogDirect(spell_id = i.spell_id, worker_id = i.worker_id, host_id = i.host_id, logs = get_res.logs, worker_name = w.name)
+                logs <<- LogsDirectResult(success = get_res.success, error = ?[get_res.error], spell_id = i.spell_id, worker_id = i.worker_id, host_id = i.host_id, logs = get_res.logs, worker_name = w.name)
 
     <- logs
 
-data LogDeal:
-    spell_id: string
-    worker_id: string
+data LogsDeal:
+    success: bool
+    error: ?string
     host_id: string
-    deal_id: string
+    worker_id: ?string
+    spell_id: ?string
     logs: []Log
 
-func get_logs_deal(deals: []string) -> *LogDeal:
-    logs: *LogDeal
+data LogsDealResult:
+    deal_id: string
+    success: bool
+    error: ?string
+    logs: []LogsDeal
+
+func get_logs_deal(deals: []string) -> []LogsDealResult:
+    result: *LogsDealResult
 
     on HOST_PEER_ID:
         for deal_id <- deals:
-            result <- Subnet.resolve(deal_id)
-            if result.success == false:
+            subnet_res <- Subnet.resolve(deal_id)
+            if !subnet_res.success:
                 on INIT_PEER_ID via HOST_PEER_ID:
-                    co Console.print(["Failed to resolve subnet for deal_id: ", deal_id, " error: ", result.error])
+                    result <<- LogsDealResult(deal_id = deal_id, success = false, error = subnet_res.error, logs = nil)
```

Comment:
what will be in the error message here?

Reply:
I am also curious how should I display it. We have nested kind of errors here.
In the test should I check for existance of my error message?
e.g. if I display this like that:

`Couldn't resolve subnet. Error: <subnet_res.error>`

in tests I would check for `Couldn't resolve subnet. Error:` string and if exists I will fail test. Sounds ok?

Reply:
Now it looks like: `Failed to resolve subnet: .....`